Session-6/DeckOfCards/CardsDeck.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DeckOfCards
+{
+    public abstract class CardsDeck<T>
+    {
+        public List<T> cardsDeck;
+        public CardsDeck()
+        {
+            cardsDeck = new List<T>();
+        }
+        public abstract void CreateDeck();
+
+        public void ShuffleDeck()
+        {
+            Random random = new Random();
+            for(int i = 0; i < cardsDeck.Count(); i++)
+            {
+                int r = random.Next(0, cardsDeck.Count());
+                T temp = cardsDeck[i];
+                cardsDeck[i] = cardsDeck[r];
+                cardsDeck[r] = temp;
+            }
+        }
+
+        public abstract void SortDeck();
+
+        public T GetTopCard()
+        {
+            return cardsDeck[0];
+        }
+        public abstract void CombineDecks(IEnumerable<dynamic> combineDeck);
```

Comment:
1. Combine deck should accept deck as input parameter
2. Combine deck should return new deck
3. You can define combining logic here
4. Implement operator overloading for deck combining. so if user wants to combine, he will simply do like *var newDeck = deck1 + deck2*